src/macos.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::{anyhow, Result};
+use security_framework::{
+    certificate::SecCertificate,
+    item::{
+        add_item, AddRef, ItemAddOptions, ItemAddValue, ItemClass, ItemSearchOptions, Reference,
+        SearchResult,
+    },
+};
+use tokio::process::Command;
+
+pub fn add_trusted_cert(der: Vec<u8>) -> Result<()> {
+    let cert = SecCertificate::from_der(&der)?;
+    let add_ref = AddRef::Certificate(cert.clone());
+    let add_option = ItemAddOptions::new(ItemAddValue::Ref(add_ref))
+        .set_label("mitmproxy")
+        .to_dictionary();
+
+    let search_result = ItemSearchOptions::new()
+        .class(ItemClass::certificate())
+        .load_refs(true)
+        .label("mitmproxy")
+        .search()
+        .map_err(|e| anyhow!(e))?;
+
+    if let Some(search_result) = search_result.first() {
+        if let SearchResult::Ref(Reference::Certificate(cert)) = search_result {
+            cert.delete()?;
+        }
+    }
+
+    add_item(add_option)?;
+
+    Command::new("open")
+        .arg("../macos-add-trusted-cert/macos-add-trusted-cert.app")
```

Comment:
This is great for local testing, but that won't work on users' machines. We need to

1) Copy the compiled app into the wheel (in build.rs).
2) Add an argument to `add_trusted_cert` to accept a path for the binary.
3) Add a wrapper in `ffi` to determine the path on the user's machine. You can pretty much copy this: https://github.com/mitmproxy/mitmproxy_rs/blob/da1000e004b645d58ac51cd98329ec7d3d711216/ffi/src/server.rs#L282-L293 :)

Reply:
ok, perfect!